src/cli/cli.go

```diff
@@ -29,9 +29,28 @@ import (
 )
 
 const (
-	zdnsCLIVersion = "1.1.0"
+	zdnsCLIVersion  = "1.1.0"
+	ModulesColWidth = 14
+	ModulesPerRow   = 6
 )
 
+// Unfortunately, we can't use the moduleToLookupModule map here, as it's not available at runtime yet.
+// Variables get initialized and then init() gets run (where modules are registered), so we'll have to hardcode the list.
+var allModules = []string{
+	"A", "AAAA", "AFSDB", "ANY", "ATMA", "AVC", "AXFR", "BINDVERSION", "CAA", "CDNSKEY", "CDS", "CERT",
+	"CNAME", "CSYNC", "DHCID", "DMARC", "DNAME", "DNSKEY", "DS", "EID", "EUI48", "EUI64", "GID", "GPOS",
+	"HINFO", "HIP", "HTTPS", "ISDN", "KEY", "KX", "L32", "L64", "LOC", "LP", "MB", "MD", "MF", "MG",
+	"MR", "MX", "MXLOOKUP", "NAPTR", "NID", "NIMLOC", "NINFO", "NS", "NSAPPTR", "NSEC", "NSEC3", "NSEC3PARAM",
+	"NSLOOKUP", "NULL", "NXT", "OPENPGPKEY", "PTR", "PX", "RP", "RRSIG", "RT", "SMIMEA", "SOA", "SPF", "SRV",
+	"SSHFP", "SVCB", "TALINK", "TKEY", "TLSA", "TXT", "UID", "UINFO", "UNSPEC", "URI",
+}
+
+// cmds is a set of commands, special "modules" with their own flags. They should not be printed as "modules" or used with --module
+var cmds = map[string]struct{}{
+	"MXLOOKUP": {},
+	"NSLOOKUP": {},
+}
+
```

Comment:
I don't understand why they shouldn't be printed as part of modules. It seems like we want the user to know they exist? They _are_ modules.

Reply:
They're modules in the way we mean it with ZDNS but in terms of Cobra/the CLI, they're commands. We register them as "commands" and they get printed by Cobra here:
```
$ ./zdns --help
...
Usage:
  zdns [flags]
  zdns [command]

Available Commands:
  completion  Generate the autocompletion script for the specified shell
  help        Help about any command
  mxlookup    Run a more exhaustive mxlookup
  nslookup    Run a more exhaustive nslookup
```

So they **must** be called with `./zdns mxlookup`., or `./zdns_binary CMD_NAME` We can't have `./zdns --module=mxlookup` since Cobra only initializes the command specific flags (`--mx-cache-size` [here](https://github.com/zmap/zdns/blob/main/src/cli/mxlookup.go#L43)) if the command is used as cobra expects.

So I'd say they are modules in the ZDNS sense but we must call them like Cobra expects commands to be called. Cobra will print out `mxlookup` and `nslookup` in the `Available Commands` section, and we'll print out the other ZDNS modules in the new section of `--help`, Available modules:
```
Available modules:
A             AAAA          AFSDB         ANY           ATMA          AVC           
AXFR          BINDVERSION   CAA           CDNSKEY       CDS           CERT          
CNAME         CSYNC         DHCID         DMARC         DNAME         DNSKEY        
DS            EID           EUI48         EUI64         GID           GPOS          
HINFO         HIP           HTTPS         ISDN          KEY           KX            
L32           L64           LOC           LP            MB            MD            
MF            MG            MR            MX            NAPTR         NID           
NIMLOC        NINFO         NS            NSAPPTR       NSEC          NSEC3         
NSEC3PARAM    NULL          NXT           OPENPGPKEY    PTR           PX            
RP            RRSIG         RT            SMIMEA        SOA           SPF           
SRV           SSHFP         SVCB          TALINK        TKEY          TLSA          
TXT           UID           UINFO         UNSPEC        URI
```

In this way, the user will know about all available options.

Reply:
Is there a reason to not just define our own `--help` that patches this up? Does cobra allow us to define that ourselves?

Reply:
I'm not really sure how that would solve things. 

```
func init() {
	rootCmd.AddCommand(mxlookupCmd)

	mxlookupCmd.PersistentFlags().Bool("ipv4-lookup", false, "perform A lookups for each MX server")
	mxlookupCmd.PersistentFlags().Bool("ipv6-lookup", false, "perform AAAA record lookups for each MX server")
	mxlookupCmd.PersistentFlags().Int("mx-cache-size", 1000, "number of records to store in MX -> A/AAAA cache")

	util.BindFlags(mxlookupCmd, viper.GetViper(), util.EnvPrefix)
}
```
The `--ipv4-lookup` and `--ipv6-lookup` don't matter too much because we also define those on the `rootCmd`, but the `mx-cache-size` is only defined here.

Cobra will only initialize that flag if the `mxlookup` command is used as a "Cobra command", ie. `zdns mxlookup`. 

Unless we define `--mx-cache-size` on the root command and let it be used by all modules, we must treat it separately because Cobra handles it differently.